pkg/apis/triggers/v1alpha1/event_listener_validation.go

```diff
@@ -43,9 +44,105 @@ func (s *EventListenerSpec) validate(ctx context.Context) *apis.FieldError {
 			return err
 		}
 	}
+	if s.Resources.KubernetesResource != nil {
+		return validateKubernetesObject(s.Resources.KubernetesResource)
+	}
 	return nil
 }
 
+func validateKubernetesObject(orig *KubernetesResource) *apis.FieldError {
+	var errs *apis.FieldError
+	if len(orig.Template.Spec.Containers) > 1 {
+		errs = errs.Also(apis.ErrMultipleOneOf("containers").ViaField("spec.template.spec"))
+	}
+	errs = errs.Also(apis.CheckDisallowedFields(orig.Template.Spec,
+		*podSpecMask(&orig.Template.Spec)).ViaField("spec.template.spec"))
+
+	// bounded by condition because containers fields are optional so there is a chance that containers can be nil.
+	if len(orig.Template.Spec.Containers) == 1 {
+		errs = errs.Also(apis.CheckDisallowedFields(orig.Template.Spec.Containers[0],
+			*containerFieldMask(&orig.Template.Spec.Containers[0])).ViaField("spec.template.spec.containers[0]"))
+	}
+
+	return errs
+}
+
+func containerFieldMask(in *corev1.Container) *corev1.Container {
+	out := new(corev1.Container)
+
+	// Allowed fields
+	out.Env = in.Env
+
+	// Disallowed fields
+	// This list clarifies which all container attributes are not allowed.
+	out.Name = ""
+	out.Image = ""
+	out.Args = nil
+	out.Ports = nil
+	out.LivenessProbe = nil
+	out.ReadinessProbe = nil
+	out.StartupProbe = nil
+	out.Command = nil
+	out.VolumeMounts = nil
+	out.ImagePullPolicy = ""
+	out.Lifecycle = nil
+	out.SecurityContext = nil
+	out.Stdin = false
+	out.StdinOnce = false
+	out.TerminationMessagePath = ""
+	out.TerminationMessagePolicy = ""
+	out.WorkingDir = ""
+	out.TTY = false
+	out.VolumeDevices = nil
+	out.EnvFrom = nil
+	out.Resources = corev1.ResourceRequirements{}
```

Comment:
We might want to allow users to set resource requirements for the container?
(to be clear, this would be a new feature so it can come later!)

Reply:
Ya that make sense :+1: 

We can support based on usecase

Shall i open a new issue so that it will be tracked ?

Reply:
yeah we can open an issue for it.

Reply:
Created issue https://github.com/tektoncd/triggers/issues/762